### PR TITLE
chore(badge): convert enzyme tests to RTL

### PR DESCRIPTION
**Pull request by nuria1110**
### Proposed behaviour

Convert `Badge` tests to RTL.

### Current behaviour

`Badge` tests are Enzyme.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [x] Unit tests added or updated if required
- [ ] Playwright automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [ ] Tested in provided StackBlitz sandbox/Storybook
- [ ] Add new Playwright test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### CODE REVIEW

File: src/components/badge/badge.test.tsx
Comment (edleeks87) [inline src/components/badge/badge.test.tsx:37]: comment: I don't think you need both assertions here, the first one suffices imo

File: src/components/badge/badge.test.tsx
Comment (edleeks87) [inline src/components/badge/badge.test.tsx:6]:
```suggestion
  render(<Badge {...props}>Foo</Badge>);
```

non-blocking just looks a bit weird having that as the child text imo

File: src/components/badge/badge.test.tsx
Comment (edleeks87) [reply src/components/badge/badge.test.tsx:9]: Also most of these don't need to be `async` tests as we're using synchronous `getBy` queries instead of `findBy` which is asynchronous and needs awaiting

File: src/components/badge/badge.test.tsx
Comment (edleeks87) [inline src/components/badge/badge.test.tsx:49]: I wonder if the fact the close icon is also rendered here and includes a pseudo that it's appearing hidden to RTL 🤷.  If we end up needing to do an investigation separate from this ticket then we should raise one and include it in this comment as a FIXMe

File: src/components/badge/badge.test.tsx
Comment (edleeks87) [inline src/components/badge/badge.test.tsx:61]: question: should we also test that it's not set when onClick is passed?

File: src/components/badge/badge.test.tsx
Comment (edleeks87) [inline src/components/badge/badge.test.tsx:67]: suggestion: I think we should avoid nesting the `describe` here, the `it` blocks have enough of a description imo

File: src/components/badge/badge.test.tsx
Comment (Parsium) [reply src/components/badge/badge.test.tsx:49]: After some investigation, removing `display: none;` CSS rule on Line 75 in `badge.style.ts` causes the test to pass. Sounds to me like either the pseudoelement @edleeks87 mentioned is causing unintended effects, or JSDOM, or even both. Worth us looking into in the future for sure :+1:

@nuria1110 would you be happy to raise a ticket for this?

File: src/components/badge/badge.test.tsx
Comment (nuria1110) [reply src/components/badge/badge.test.tsx:9]: Yeah I'll remove both of those tests, they don't seem to affect coverage

File: src/components/badge/badge.test.tsx
Comment (Parsium) [inline src/components/badge/badge.test.tsx:6]: **comment**: just wanted to mention, this simple abstraction is alright for now, but let's keep an eye out for similar ones in future PRs, especially when we're dealing with more complex test files.

It's important to ensure that any helper functions we add genuinely enhance clarity and don't make the file structure more complicated in the name of staying DRY.

**Comment by Parsium** [comment]
Not much to add to @edleeks87's comments - nice work @nuria1110 👍🏼

File: src/components/badge/badge.test.tsx
Comment (nuria1110) [reply src/components/badge/badge.test.tsx:49]: Raised FE-6575 to investigate this.

**Review by Parsium — APPROVED** [review]
LGTM @nuria1110 - once @edleeks87 agrees let's squash the changes 👍🏼

**Review by edleeks87 — APPROVED** [review]

**Comment by edleeks87** [comment]
Looks good @nuria1110, yeah lets squash the commits before we merge but no rush 👍

**Review by ZhuoyuJin — APPROVED** [review]

**Pull request merged**
